File: leetcode75/03-1431. Kids With the Greatest Number of Candies/main.py
"""
1. Find the greatest number in the array
2. For  each integer in array, add the extra candies,  check if it is greater than or equals to the greatest.
3. Set values accordingly
"""


class Solution:
    def kidsWithCandies(self, candies: list[int], extraCandies: int) -> list[bool]:
        greatest = max(candies)

        result = [False] * len(candies)

        for index, value in enumerate(candies):
            if value + extraCandies >= greatest:
                result[index] = True
        return result

File: leetcode75/03-1431. Kids With the Greatest Number of Candies/main.rs
use std::cmp::max;
impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
        let greatest = candies.iter().max().unwrap();

        let mut result = vec![false; candies.len()];

        for (index, value) in candies.iter().enumerate() {
            if *value + extra_candies > *greatest {
                result[index] = true;
            }
        }
        return result;
    }
}

pub fn optimized(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
    if let Some(&greatest) = candies.iter().max() {
        candies
            .iter()
            .map(|&value| value + extra_candies >= greatest)
    } else {
        Vec::new()
    }
}
